ocaml/typing/types.mli

```diff
@@ -955,3 +964,21 @@ val set_univar: type_expr option ref -> type_expr -> unit
 val link_kind: inside:field_kind -> field_kind -> unit
 val link_commu: inside:commutable -> commutable -> unit
 val set_commu_ok: commutable -> unit
+
+(*  *)
+module AppArgs : sig
+  type t = app_args
+
+  val unapp : t
```

Comment:
My intention (and the current state) is that it has `private` constructors. But perhaps that doesn't make sense.

Reply:
I hadn't spotted that. I guess you're trying to ensure that the `Applied` case has a non-empty list? Personally, I don't think it is worth it. That invariant isn't really needed semantically (`type () t` would make sense it's just useless), and you're losing the nice constructor disambiguation that let's you just write `Tconstr(p, Unapplied)` instead of `Tconstr(p, Types.unapp)`.